cdap-watchdog/src/main/java/io/cdap/cdap/logging/plugins/RollingLocationLogAppender.java

```diff
@@ -206,39 +210,10 @@ public void sync() throws IOException {
 
   @Override
   public void stop() {
-    try {
-      LOG.info("Stopping appender {}", this.name);
-      locationManager.close();
-      if (encoder != null) {
-        encoder.close();
-      }
-    } catch (IOException ioe) {
-      LOG.error("Failed to write footer for appender named {}", this.getName(), ioe);
-    } finally {
-      this.started = false;
-    }
-  }
+    LOG.info("Stopping appender {}", this.name);
+    super.stop();
+    locationManager.close();
```

Comment:
do we need to put this in a try-finally or will super.stop() never throw an exception?

Reply:
also, do we need to close the delegatingOutputStream?

Reply:
super.stop() has a try-catch block that catches any exception that occurs. So calling that seems sufficient. I have called locationManager.close() that closes all the delegatingOutputStreams that are open. I consulted with @tivv and as per our discussion, it was decided that it was the right thing to do.